File: src/main/java/com/extendedae_plus/mixin/impl/ExclusionZoneScalingButton.java
package com.extendedae_plus.mixin.impl;

import com.extendedae_plus.mixin.impl.bridge.HelperProviderButtons;
import dev.emi.emi.api.EmiExclusionArea;
import dev.emi.emi.api.widget.Bounds;
import net.minecraft.client.gui.screens.Screen;

import java.util.function.Consumer;

public class ExclusionZoneScalingButton implements EmiExclusionArea<Screen> {
    @Override
    public void addExclusionArea(Screen screen, Consumer<Bounds> consumer) {
        if (!(screen instanceof HelperProviderButtons helper)) return;

        var buttons = helper.eaep$getButtons();
        if (buttons.isEmpty()) return;

        int minX = Integer.MAX_VALUE;
        int minY = Integer.MAX_VALUE;
        int maxX = Integer.MIN_VALUE;
        int maxY = Integer.MIN_VALUE;

        for (var button : buttons) {
            if (button.getAction() == null || !"scaling".equals(button.getAction().getGroup())) continue;

            int buttonX = button.getX();
            int buttonY = button.getY();
            int buttonWidth = button.getWidth();
            int buttonHeight = button.getHeight();

            minX = Math.min(minX, buttonX);
            minY = Math.min(minY, buttonY);
            maxX = Math.max(maxX, buttonX + buttonWidth);
            maxY = Math.max(maxY, buttonY + buttonHeight);
        }

        int width = maxX - minX;
        int height = maxY - minY;
        consumer.accept(new Bounds(minX, minY, width, height));
    }
}

File: src/main/kotlin/com/extendedae_plus/mixin/impl/ExclusionZoneScalingButton.kt
package com.extendedae_plus.mixin.impl

import com.extendedae_plus.mixin.helper.HelperProviderButtons
import dev.emi.emi.api.EmiExclusionArea
import dev.emi.emi.api.widget.Bounds
import net.minecraft.client.gui.screens.Screen
import java.util.function.Consumer
import kotlin.math.max
import kotlin.math.min

object ExclusionZoneScalingButton : EmiExclusionArea<Screen> {
    override fun addExclusionArea(screen: Screen, consumer: Consumer<Bounds>) {
        if (screen !is HelperProviderButtons) return

        val buttons = screen.`eaep$getButtons`()
        if (buttons.isEmpty()) return

        var minX = Int.MAX_VALUE
        var minY = Int.MAX_VALUE
        var maxX = Int.MIN_VALUE
        var maxY = Int.MIN_VALUE

        for (button in buttons) {
            if (button.action == null || "scaling" != button.action?.group) continue

            val buttonX = button.x
            val buttonY = button.y

            minX = min(minX, buttonX)
            minY = min(minY, buttonY)
            maxX = max(maxX, buttonX + button.width)
            maxY = max(maxY, buttonY + button.height)
        }

        val width = maxX - minX
        val height = maxY - minY
        consumer.accept(Bounds(minX, minY, width, height))
    }
}
